### PR TITLE
Feature/allow pics with camera

### LINKED ISSUE #85: 02.03.02 Allow user to take picture with camera for habit events
**User statement**
As a doer, I want the photograph for a habit event to be taken with the camera within the app.

**Describe the solution you'd like**
Add popup that lets user choose between gallery or camera, let them take picture with camera

**Development notes**

### PR DESCRIPTION
**Pull request by JuliePilz**
### What is it?
This PR allows a user to takes pictures with their camera when adding pictures to habit events or their profile

Adding to profile:
Screenshot_20211124-215800_HabitSmasher
Screenshot_20211124-215811_Camera
Screenshot_20211124-220449_HabitSmasher

Adding to Habit Event:
Screenshot_20211124-220619_HabitSmasher
Screenshot_20211124-220653_HabitSmasher
Screenshot_20211124-220717_HabitSmasher

Closes #85 

### Tasks

- [x] run unit tests
- [x] run UI tests

### CODE REVIEW

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+                // If the user takes the image it will already be stored in _selectedImage
+                if (resultCode == RESULT_OK) {
+                    // Set selected picture
+                    _parentDialog.setImage(_selectedImage);
+
+                    // Close dialog
+                    this.dismiss();
+                }
+                break;
+        }
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:124]: Just a question, if we are doing conditional checks on the `resultCode`, why do we need the `switch-case` statement? Can we not just go right to the `if` statements?

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+    /**
+     * Creates a temporary image file when the camera is used
+     * Reference: https://developer.android.com/training/camera/photobasics#java
+     * @return The created image
+     * @throws IOException
+     */
+    private File createImageFile() throws IOException {
+        // Create an image file name
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:193]: Minor: could extract the data format into a constant

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+        String imageFileName = "JPEG_" + timeStamp + "_";
+        File storageDir = getContext().getExternalFilesDir(Environment.DIRECTORY_PICTURES);
+        File image = File.createTempFile(
+                imageFileName,  /* prefix */
+                ".jpg",         /* suffix */
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:198]: Minor: could extract the file extension into a constant

File: app/src/main/java/com/example/habitsmasher/ImageDatabaseHelper.java [hunk 4]
```diff
@@ -19,7 +19,7 @@
     private StorageReference _storageReference = _storage.getReference();
 
     // Max download megabyte size
-    private final long ONE_MEGABYTE = 1024 * 1024;
+    private final long ONE_MEGABYTE = 1024 * 1024 * 5;
```
Comment (rudy-patel) [inline app/src/main/java/com/example/habitsmasher/ImageDatabaseHelper.java:22]: Is this no longer `ONE_MEGABYTE`? Is it now `FIVE_MEGABYTES`?

**Review by rudy-patel — DISMISSED** [review]
Looks good. The code is clean and well documented. Left a couple of minor thoughts/comments, up to you if you want to implement those. Approved.

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java (hunk 1 above)
Comment (JuliePilz) [reply app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:124]: The switch-case statement is for the requestCode (slightly different). The first case is essentially the gallery case, while the second one is the camera case. Both still require the result code to be RESULT_OK though

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java (hunk 2 above)
Comment (JuliePilz) [reply app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:193]: Good idea :)

File: app/src/main/java/com/example/habitsmasher/ImageDatabaseHelper.java (hunk 4 above)
Comment (JuliePilz) [reply app/src/main/java/com/example/habitsmasher/ImageDatabaseHelper.java:22]: Good catch

File: app/src/main/java/com/example/habitsmasher/AddPictureDialog.java (hunk 1 above)
Comment (rudy-patel) [reply app/src/main/java/com/example/habitsmasher/AddPictureDialog.java:128]: ahhh, apparently I can't read. That makes sense, thanks.

**Review by rudy-patel — APPROVED** [review]

**Review by rudy-patel — APPROVED** [review]

**Review by jaskim9824 — APPROVED** [review]

**Pull request merged**